src/BlockchainWriter/Controller.ts

```diff
@@ -90,8 +91,19 @@ export class Controller {
     })
   }
 
-  private anchorData = async (data: string) => {
+  private fundTransaction = async (rawTransaction: string) => {
```

Comment:
This can be simplified by using [`rethrow`](https://github.com/poetapp/node/blob/master/src/Extensions/Promise.ts#L6):

```suggestion
// outside the class, as a pure function
const translateFundTransactionError = async (error: Error) => 
  exception.message === 'Insufficient funds'
    ? new InsufficientFundsException('Insuffiecient funds for transaction')
    : error
```

```ts
// in the class
private anchorData = async (data: string) => {
  /// ...
  const fundedTransaction = await bitcoinCore.fundTransaction(rawTransaction).rethrow(translateFundTransactionError)
```

Reply:
I'm getting this error when trying to use rethrow: 
```
module: "BlockchainWriter"
    file: "Controller"
    method: "anchorNextHash"
    ipfsDirectoryHash: "QmWy3kwndywtopWB2audW12HFnAe2gcQe3MupdSV8dNKX1"
    exception: {
      "stack": [
        "TypeError: bitcoinCore.fundRawTransaction(...).rethrow is not a function",
        "    at Controller.anchorData (/usr/src/app/dist/babel/src/BlockchainWriter/Controller.js:36:92)",
        "    at process._tickCallback (internal/process/next_tick.js:68:7)"
      ],
      "message": "bitcoinCore.fundRawTransaction(...).rethrow is not a function",
      "type": "TypeError"
    }
```

Reply:
@jwicks31 I think you might need to import the rethrow promise extensions. src/Extensions/Promise.ts

Reply:
It looks like they are imported at the app level. Should that cover it?

Reply:
Ah, yeah if its already imported then you shouldn't need to import again. Can you find another example of rethrow somewhere else being used in the codebase?

Reply:
`src/StorageReader/ClaimController.ts` uses it.

Reply:
Interesting, maybe bitcoinCore.fundRawTransaction does not return an es6 Promise, but a promise from a custom library or something, that is the only thing I can think of that the global es6 Promise is not being extended.

Reply:
Maybe wrap the call with another promise so you can use rethrow or wrap it with another function thats only responsibility is the try/catch for now. I don't have time to investigate why rethrow is not working on bitcoinCore.fundRawTransaction

Reply:
Darn, it uses bluebird. 

We can refactor `rethrow` in the future so it takes a Promise in a curried way and can be used in pipes.

For now I think this is best: 

```ts
const translateFundTransactionError = (error: Error) => {
  throw error.message === 'Insufficient funds'
    ? new InsufficientFundsException('Insuffiecient funds for transaction')
    : error
}

const fundedTransaction = await bitcoinCore.fundRawTransaction(rawTransaction).catch(translateFundTransactionError)

```